### PR TITLE
Add initial implementation for common controller and tfstate parsing&conversion

### PR DESCRIPTION
**Pull request by turkenh**
This PR adds an initial implementation (subject to change) for Terraform common controller interacting with the Scheduler interface. 

It also adds support parse tfstate and primitives to do conversions.

I'll open a follow-up pr using this common controller in https://github.com/crossplane-contrib/provider-tf-aws repo.

### CODE REVIEW

File: pkg/reconciler/external.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	dr, err := e.executor.Delete(ctx, tr)
+	if err != nil {
+		return errors.Wrap(err, "failed to delete")
+	}
+	if dr.Completed {
+		return nil
+	}
+	return errors.Errorf("still deleting")
```
Comment (negz) [inline pkg/reconciler/external.go:126]: What's going on here? Do we need to wait? 🤔

File: pkg/reconciler/external.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+func (e *TerraformExternal) Update(ctx context.Context, mg xpresource.Managed) (managed.ExternalUpdate, error) {
+	tr, ok := mg.(resource.Terraformed)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	ar, err := e.executor.Apply(ctx, tr)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, "failed to apply")
+	}
```
Comment (negz) [inline pkg/reconciler/external.go:106]: Do we need to wait for `ar.Completed`?

File: pkg/conversion/statev4.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
+package conversion
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"github.com/pkg/errors"
+)
+
+const (
+	AnnotationKeyStateMeta = "tf.crossplane.io/state-meta"
```
Comment (negz) [inline pkg/conversion/statev4.go:10]: Is this file excluded from linting? It seems like it would fail due to missing commentary.

**Comment by turkenh** [comment]
Thanks @negz for the review but this PR is intended to be a draft (just converted) to sync on a high-level structure.
I'll apply your comments for the actual PR if applicable.

**Comment by turkenh** [comment]
Closing in favor of #14

**Pull request closed without merging**